frontend/src/container/GridCardLayout/GridCard/index.tsx

```diff
@@ -0,0 +1,130 @@
+import Spinner from 'components/Spinner';
+import { PANEL_TYPES } from 'constants/queryBuilder';
+import { useGetQueryRange } from 'hooks/queryBuilder/useGetQueryRange';
+import { useStepInterval } from 'hooks/queryBuilder/useStepInterval';
+import { getDashboardVariables } from 'lib/dashbaordVariables/getDashboardVariables';
+import getChartData from 'lib/getChartData';
+import isEmpty from 'lodash-es/isEmpty';
+import { useDashboard } from 'providers/Dashboard/Dashboard';
+import { memo, useMemo, useState } from 'react';
+import { useInView } from 'react-intersection-observer';
+import { useDispatch, useSelector } from 'react-redux';
+import { UpdateTimeInterval } from 'store/actions';
+import { AppState } from 'store/reducers';
+import { GlobalReducer } from 'types/reducer/globalTime';
+
+import EmptyWidget from '../EmptyWidget';
+import { MenuItemKeys } from '../WidgetHeader/contants';
+import { GridCardGraphProps } from './types';
+import WidgetGraphComponent from './WidgetGraphComponent';
+
+function GridCardGraph({
+	widget,
+	name,
+	onClickHandler,
+	headerMenuList = [MenuItemKeys.View],
+	isQueryEnabled,
+	threshold,
+}: GridCardGraphProps): JSX.Element {
+	const dispatch = useDispatch();
+	const [errorMessage, setErrorMessage] = useState<string>();
+
+	const onDragSelect = (start: number, end: number): void => {
+		const startTimestamp = Math.trunc(start);
+		const endTimestamp = Math.trunc(end);
+
+		if (startTimestamp !== endTimestamp) {
+			dispatch(UpdateTimeInterval('custom', [startTimestamp, endTimestamp]));
+		}
+	};
+
+	const { ref: graphRef, inView: isGraphVisible } = useInView({
+		threshold: 0,
+		triggerOnce: true,
+		initialInView: false,
+	});
+
+	const { selectedDashboard } = useDashboard();
+
+	const { minTime, maxTime, selectedTime: globalSelectedInterval } = useSelector<
+		AppState,
+		GlobalReducer
+	>((state) => state.globalTime);
+
+	const updatedQuery = useStepInterval(widget?.query);
+
+	const isEmptyWidget =
+		widget?.id === PANEL_TYPES.EMPTY_WIDGET || isEmpty(widget);
+
+	const queryResponse = useGetQueryRange(
+		{
+			selectedTime: widget?.timePreferance,
+			graphType: widget?.panelTypes,
+			query: updatedQuery,
+			globalSelectedInterval,
+			variables: getDashboardVariables(selectedDashboard?.data.variables),
```

Comment:
Default the param to default variables type

Reply:
already handled in the function